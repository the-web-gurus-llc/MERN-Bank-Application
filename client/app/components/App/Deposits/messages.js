import { defineMessages } from 'react-intl';

export const scope = 'app.components.Deposits';

export default defineMessages({
  title: {
    id: `${scope}.title`,
    defaultMessage: 'Deposits',
  },
  description: {
    id: `${scope}.description`,
    defaultMessage: 'The deposits function is disabled.',
  },
  action: {
    id: `${scope}.action`,
    defaultMessage: 'New deposit',
  },
});
