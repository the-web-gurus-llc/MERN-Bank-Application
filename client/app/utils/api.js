const API_BASE_URL = 'https://api.pietrzakadrian.com/bank';
const AUTH = 'Auth';
const USERS = 'Users';
const CURRENCIES = 'Currencies';
const BILLS = 'Bills';
const TRANSACTIONS = 'Transactions';
const MESSAGES = 'Messages';
const NOTIFICATIONS = 'Notifications';

export default {
  auth: {
    login: `${API_BASE_URL}/${AUTH}/login`,
    register: `${API_BASE_URL}/${AUTH}/register`,
    logout: `${API_BASE_URL}/${AUTH}/logout`,
    forgetPassword: `${API_BASE_URL}/${AUTH}/password/forget`,
    resetPassword: `${API_BASE_URL}/${AUTH}/password/reset`,
  },
  users: (path = '') => {
    switch (path) {
      case 'checkEmail':
        return (email) => `${API_BASE_URL}/${USERS}/${email}/checkEmail`;
      default:
        return `${API_BASE_URL}/${USERS}`;
    }
  },
  currencies: `${API_BASE_URL}/${CURRENCIES}`,
  messages: `${API_BASE_URL}/${MESSAGES}`,
  notifications: `${API_BASE_URL}/${NOTIFICATIONS}`,
  bills: (path = '') => {
    switch (path) {
      case 'amountMoney':
        return `${API_BASE_URL}/${BILLS}/amountMoney`;
      case 'accountBalance':
        return `${API_BASE_URL}/${BILLS}/accountBalance`;
      case 'accountBalanceHistory':
        return `${API_BASE_URL}/${BILLS}/accountBalanceHistory`;
      case 'search':
        return (accountBillNumber) =>
          `${API_BASE_URL}/${BILLS}/${accountBillNumber}/search`;
      default:
        return `${API_BASE_URL}/${BILLS}`;
    }
  },
  transactions: (path = '') => {
    switch (path) {
      case 'create':
        return `${API_BASE_URL}/${TRANSACTIONS}/create`;
      case 'confirm':
        return `${API_BASE_URL}/${TRANSACTIONS}/confirm`;
      case 'authorizationKey':
        return (uuid) =>
          `${API_BASE_URL}/${TRANSACTIONS}/${uuid}/authorizationKey`;
      case 'confirmationFile':
        return (uuid, locale) =>
          `${API_BASE_URL}/${TRANSACTIONS}/${uuid}/${locale}/confirmationFile`;
      default:
        return `${API_BASE_URL}/${TRANSACTIONS}`;
    }
  },
};
