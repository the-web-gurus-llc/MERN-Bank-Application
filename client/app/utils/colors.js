export default {
  white: '#fff',
  black: '#000',
  primaryBlue: '#1890ff',
  grey: 'rgb(242, 244, 247)',
  red: 'rgb(229, 0, 0)',
  redErr: '#ff4d4f',
  silver: '#bababa',
};
