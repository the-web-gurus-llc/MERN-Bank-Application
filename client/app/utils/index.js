export { default as colors } from './colors';
export { default as media } from './media';
export { default as request } from './request';
export { default as typography } from './typography';
export { default as api } from './api';
export { default as routes } from './routes';
export { default as locales } from './locales';
export { default as i18n } from './i18n';
