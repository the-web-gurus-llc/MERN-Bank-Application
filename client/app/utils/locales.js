exports.DEFAULT_LOCALE = 'en';
exports.DEFAULT_DATE_FORMAT = 'dd.MM.yyyy, hh:mm aa';

// prettier-ignore
exports.appLocales = [
  'en',
  'de',
  'pl',
];
