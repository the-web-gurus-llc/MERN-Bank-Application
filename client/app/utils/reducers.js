/**
 * Combine all reducers in this file and export the combined reducers.
 */

import { combineReducers } from 'redux';
import { connectRouter } from 'connected-react-router';

import history from 'utils/history';
import globalReducer from 'containers/App/reducer';
import languageProviderReducer from 'providers/LanguageProvider/reducer';
import loadingProviderReducer from 'providers/LoadingProvider/reducer';
import errorProviderReducer from 'providers/ErrorProvider/reducer';

/**
 * Merges the main reducer with the router state and dynamically injected reducers
 */
export default function createReducer(injectedReducers = {}) {
  const rootReducer = combineReducers({
    global: globalReducer,
    language: languageProviderReducer,
    loading: loadingProviderReducer,
    error: errorProviderReducer,
    router: connectRouter(history),
    ...injectedReducers,
  });

  return rootReducer;
}
