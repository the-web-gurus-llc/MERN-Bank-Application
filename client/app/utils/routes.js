import React from 'react';
import registerMessages from 'containers/RegisterPage/messages';
import loginMessages from 'containers/LoginPage/messages';
import dashboardMessages from 'containers/DashboardPage/messages';
import paymentMessages from 'containers/PaymentPage/messages';
import historyMessages from 'containers/HistoryPage/messages';
import settingsMessages from 'containers/SettingsPage/messages';
import privacyMessages from 'containers/PrivacyPage/messages';
import forgetPasswordMessages from 'containers/ForgetPasswordPage/messages';
import resetPasswordMessages from 'containers/ResetPasswordPage/messages';

import { FormattedMessage } from 'react-intl';

export default {
  home: {
    name: 'Home',
    path: '/',
  },
  login: {
    name: <FormattedMessage {...loginMessages.login} />,
    path: '/login',
  },
  register: {
    name: <FormattedMessage {...registerMessages.registration} />,
    path: '/register',
  },
  dashboard: {
    name: <FormattedMessage {...dashboardMessages.dashboard} />,
    path: '/dashboard',
  },
  payment: {
    name: <FormattedMessage {...paymentMessages.payment} />,
    path: '/payment',
  },
  history: {
    name: <FormattedMessage {...historyMessages.history} />,
    path: '/history',
  },
  settings: {
    name: <FormattedMessage {...settingsMessages.settings} />,
    path: '/settings',
  },
  privacy: {
    name: <FormattedMessage {...privacyMessages.privacy} />,
    path: '/privacy',
  },
  forgetPassword: {
    name: <FormattedMessage {...forgetPasswordMessages.forgetPassword} />,
    path: '/password/forget',
  },
  resetPassword: {
    name: <FormattedMessage {...resetPasswordMessages.resetPassword} />,
    path: '/password/reset',
  },
  notFound: { name: '404', path: '/404' },
};
