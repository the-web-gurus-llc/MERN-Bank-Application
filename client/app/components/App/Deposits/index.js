import React from 'react';
import { FormattedMessage } from 'react-intl';
import { BankOutlined } from '@ant-design/icons';
import { StyledCard, StyledCardContent } from 'components/App/Card/styles';
import {
  StyledButtonContent,
  StyledButton,
} from 'components/App/Button/styles';
import messages from './messages';

export default function Deposits() {
  return (
    <FormattedMessage {...messages.title}>
      {(title) => (
        <StyledCard
          darker="true"
          bordered="true"
          title={title}
          excluded="true"
          extra={
            <StyledButton type="link">
              <StyledButtonContent onMouseDown={(e) => e.stopPropagation()}>
                <BankOutlined /> <FormattedMessage {...messages.action} />
              </StyledButtonContent>
            </StyledButton>
          }
          shadowed="true"
        >
          <StyledCardContent onMouseDown={(e) => e.stopPropagation()}>
            <FormattedMessage {...messages.description} />
          </StyledCardContent>
        </StyledCard>
      )}
    </FormattedMessage>
  );
}
