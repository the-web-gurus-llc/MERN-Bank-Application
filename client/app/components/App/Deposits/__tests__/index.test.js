import React from 'react';
import { render } from '@testing-library/react';
import { IntlProvider } from 'react-intl';
import { DEFAULT_LOCALE } from 'utils/locales';
import Deposits from '../index';
import 'utils/__tests__/__mocks__/matchMedia';

describe('<Deposits />', () => {
  it('should render a Deposits', () => {
    const { container } = render(
      <IntlProvider locale={DEFAULT_LOCALE}>
        <Deposits />
      </IntlProvider>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
