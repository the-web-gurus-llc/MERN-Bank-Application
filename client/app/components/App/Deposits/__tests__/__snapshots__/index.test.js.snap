// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Deposits /> should render a Deposits 1`] = `
<div
  class="ant-card Cardstyle__StyledCard-s807iu-0 egrUQF ant-card-bordered"
  darker="true"
  excluded="true"
  shadowed="true"
>
  <div
    class="ant-card-head"
  >
    <div
      class="ant-card-head-wrapper"
    >
      <div
        class="ant-card-head-title"
      >
        Deposits
      </div>
      <div
        class="ant-card-extra"
      >
        <button
          class="ant-btn Buttonstyle__StyledButton-nh59mr-0 bcqyjQ ant-btn-link"
          type="button"
        >
          <div
            class="Buttonstyle__StyledButtonContent-nh59mr-1 cBxojI"
          >
            <span
              aria-label="bank"
              class="anticon anticon-bank"
              role="img"
            >
              <svg
                aria-hidden="true"
                class=""
                data-icon="bank"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M894 462c30.9 0 43.8-39.7 18.7-58L530.8 126.2a31.81 31.81 0 00-37.6 0L111.3 404c-25.1 18.2-12.2 58 18.8 58H192v374h-72c-4.4 0-8 3.6-8 8v52c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-52c0-4.4-3.6-8-8-8h-72V462h62zM512 196.7l271.1 197.2H240.9L512 196.7zM264 462h117v374H264V462zm189 0h117v374H453V462zm307 374H642V462h118v374z"
                />
              </svg>
            </span>
             
            <span>
              New deposit
            </span>
          </div>
        </button>
      </div>
    </div>
  </div>
  <div
    class="ant-card-body"
  >
    <div
      class="Cardstyle__StyledCardContent-s807iu-1 fnMJaW"
    >
      <span>
        The deposits function is disabled.
      </span>
    </div>
  </div>
</div>
`;
